### DISCUSSION TITLE
PhpDocInfoFactory does not find any annotations anymore

### DISCUSSION THREAD

**Discussion by shyim** [rectorphp/rector#8312] [Q&A] [answered]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.18.8   |

## Minimal PHP Code Causing Issue

```bash
git clone https://github.com/FriendsOfShopware/shopware-rector
cd shopware-rector
composer update
./vendor/bin/phpunit
```

The tests are failing with 0.18.8, as it cannot find the Annotations:

Rector: https://github.com/FriendsOfShopware/shopware-rector/blob/main/src/Rule/v65/MigrateCaptchaAnnotationToRouteRector.php#L72-L76
Fixture: https://github.com/FriendsOfShopware/shopware-rector/blob/main/tests/Rector/v65/MigrateCaptchaAnnotationToRoute/Fixture/RefactorWithDefaultsSetForRoute.php.inc

## Expected Behaviour

Tests are green as like the Rector version before

**Comment by samsonasik** [comment, marked as answer]
You need to use `Rector\Comments\NodeDocBlock\DocBlockUpdater` service to refresh node before return `$node:

```diff
diff --git a/src/Rule/v65/MigrateCaptchaAnnotationToRouteRector.php b/src/Rule/v65/MigrateCaptchaAnnotationToRouteRector.php
index fd9db4b..00809f6 100644
--- a/src/Rule/v65/MigrateCaptchaAnnotationToRouteRector.php
+++ b/src/Rule/v65/MigrateCaptchaAnnotationToRouteRector.php
@@ -12,13 +12,17 @@ use Rector\BetterPhpDocParser\PhpDoc\StringNode;
 use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
 use Rector\BetterPhpDocParser\PhpDocManipulator\PhpDocTagRemover;
 use Rector\BetterPhpDocParser\ValueObject\PhpDoc\DoctrineAnnotation\CurlyListNode;
+use Rector\Comments\NodeDocBlock\DocBlockUpdater;
 use Rector\Core\Rector\AbstractRector;
 use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
 use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
 
 class MigrateCaptchaAnnotationToRouteRector extends AbstractRector
 {
-    public function __construct(private PhpDocTagRemover $phpDocTagRemover, private PhpDocInfoFactory $phpDocFactory)
+    public function __construct(
+        private PhpDocTagRemover $phpDocTagRemover, private PhpDocInfoFactory $phpDocFactory,
+        private readonly DocBlockUpdater $docBlockUpdater
+        )
     {
     }
 
@@ -101,6 +105,8 @@ class MigrateCaptchaAnnotationToRouteRector extends AbstractRector
 
         $this->phpDocTagRemover->removeByName($phpDocInfo, 'Captcha');
 
+        $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
+
         return $node;
```

**Reply by shyim** [reply]
It exits already by getting the Annotation from the PhpDocInfo. So a lot before

**Reply by samsonasik** [reply]
I tried your repo, and using `DocBlockUpdater` is working, see the diff above, the 

```
$this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
```

ensure updated/deleted/added docblock reprinted

**Reply by samsonasik** [reply]
@shyim here I created a PR to your repo to update your code

- https://github.com/FriendsOfShopware/shopware-rector/pull/25
